src/Commands/Concerns/ManagesProcess.php

```diff
@@ -83,13 +89,43 @@ public function createPendingProcess(): PendingProcess
         return $process->env([
             'TERM' => 'xterm-256color',
             'FORCE_COLOR' => '1',
-            'COLUMNS' => $this->scrollPaneWidth(),
-            'LINES' => $this->scrollPaneHeight(),
+            'COLUMNS' => $screen->width,
+            'LINES' => $screen->height,
             ...$this->environment,
             ...$process->environment
         ]);
     }
 
+    protected function localEnvironmentVariables()
+    {
+        $locale = $this->utf8Locale();
+
+        return "export LC_ALL={$locale}; export LANG={$locale}";
+    }
+
+    protected function utf8Locale()
+    {
+        $locale = function_exists('locale_get_default')
+            ? locale_get_default()
+            : (getenv('LC_ALL') ?: (getenv('LC_CTYPE') ?: getenv('LANG')));
+
+        if (!$locale) {
+            return 'en_US.UTF-8';
+        }
+
+        if (stripos($locale, 'UTF-8') !== false) {
+            return $locale;
+        }
+
+        if (str_contains($locale, '.')) {
+            [$langRegion, $encoding] = explode('.', $locale, 2);
+
+            return $langRegion . '.UTF-8';
+        }
+
+        return $locale . '.UTF-8';
```

Comment:
Functionally identical.

```suggestion
        return explode('.', $locale, 2)[0] . '.UTF-8';
```

Reply:
Nice! Good catch!